src/services/codefixes/convertToAsyncFunction.ts

```diff
@@ -365,85 +311,65 @@ namespace ts.codefix {
+        const onFulfilledArgumentName = getArgBindingName(onFulfilled, transformer);
+        const transformationBody = getTransformationBody(onFulfilled, prevArgName, onFulfilledArgumentName, node, transformer);
 
-        if (!res) {
-            return transformExpression(node.expression, transformer, outermostParent);
-        }
-
-        const argNameRes = getArgBindingName(res, transformer);
-        const transformationBody = getTransformationBody(res, prevArgName, argNameRes, node, transformer);
-
-        if (rej) {
-            const argNameRej = getArgBindingName(rej, transformer);
-
-            const tryBlock = createBlock(transformExpression(node.expression, transformer, node, argNameRes).concat(transformationBody));
-
-            const transformationBody2 = getTransformationBody(rej, prevArgName, argNameRej, node, transformer);
-
-            const catchArg = argNameRej ? isSynthIdentifier(argNameRej) ? argNameRej.identifier.text : argNameRej.bindingPattern : "e";
+        if (onRejected) {
+            const onRejectedArgumentName = getArgBindingName(onRejected, transformer);
+            const tryBlock = createBlock(transformExpression(node.expression, transformer, onFulfilledArgumentName).concat(transformationBody));
+            const transformationBody2 = getTransformationBody(onRejected, prevArgName, onRejectedArgumentName, node, transformer);
+            const catchArg = onRejectedArgumentName ? isSynthIdentifier(onRejectedArgumentName) ? onRejectedArgumentName.identifier.text : onRejectedArgumentName.bindingPattern : "e";
             const catchVariableDeclaration = createVariableDeclaration(catchArg);
             const catchClause = createCatchClause(catchVariableDeclaration, createBlock(transformationBody2));
 
             return [createTry(tryBlock, catchClause, /* finallyBlock */ undefined)];
         }
 
-        return transformExpression(node.expression, transformer, node, argNameRes).concat(transformationBody);
-    }
-
-    function getFlagOfBindingName(bindingName: SynthBindingName, constIdentifiers: readonly Identifier[]): NodeFlags {
-        const identifiers = getIdentifierTextsFromBindingName(getNode(bindingName));
-        const inArr: boolean = constIdentifiers.some(elem => contains(identifiers, elem.text));
-        return inArr ? NodeFlags.Const : NodeFlags.Let;
+        return transformExpression(node.expression, transformer, onFulfilledArgumentName).concat(transformationBody);
     }
 
-    function transformPromiseCall(node: Expression, transformer: Transformer, prevArgName?: SynthBindingName): readonly Statement[] {
-        const shouldReturn = transformer.setOfExpressionsToReturn.get(getNodeId(node).toString());
-        // the identifier is empty when the handler (.then()) ignores the argument - In this situation we do not need to save the result of the promise returning call
-        const originalNodeParent = node.original ? node.original.parent : node.parent;

```

Comment:
This shows the confusion caused by `node.original` being missing for identifiers. There seems to be an assumption that if `originalNodeParent` was missing, it was probably a property access expression. What I found after ensuring that `node.original.parent` always exists is that it is _always_ a property access expression, so parts of the following condition were always true.

Reply:
Also, the original name was confusing—what is a “promise call”? The node passed here wasn’t even a call expression much of the time.